src/main/webapp/app/shared/import-list/import-table.component.ts

```diff
@@ -0,0 +1,119 @@
+import { Component, computed, effect, inject, input, output, signal, untracked } from '@angular/core';
+import { BaseEntity } from 'app/shared/model/base-entity';
+import { PagingService } from 'app/exercises/shared/manage/paging.service';
+import { SearchResult, SearchTermPageableSearch, SortingOrder } from 'app/shared/table/pageable-table';
+import { lastValueFrom } from 'rxjs';
+import { AlertService } from 'app/core/util/alert.service';
+import { onError } from 'app/shared/util/global.utils';
+import { ArtemisSharedCommonModule } from 'app/shared/shared-common.module';
+import { faSort, faSortDown, faSortUp, faSpinner } from '@fortawesome/free-solid-svg-icons';
+import { BaseApiHttpService } from 'app/course/learning-paths/services/base-api-http.service';
+
+/**
+ * An abstract component intended for cases where a resource needs to be imported from one course into another.
+ *
+ * @template T generic class parameter of the entity that gets imported
+ */
+export type Column<T extends BaseEntity> = {
+    name: string;
+    getProperty(entity: T): string | undefined;
+};
+
+@Component({
+    selector: 'jhi-import-table',
+    standalone: true,
+    imports: [ArtemisSharedCommonModule],
+    templateUrl: './import-table.component.html',
+    styleUrl: './import-table.component.scss',
+})
+export class ImportTableComponent<T extends BaseEntity> {
+    protected readonly SortingOrder = SortingOrder;
+
+    protected readonly faSort = faSort;
+    protected readonly faSortUp = faSortUp;
+    protected readonly faSortDown = faSortDown;
+    protected readonly faSpinner = faSpinner;
+
+    private readonly alertService = inject(AlertService);
+    private readonly pagingService = inject(PagingService);
+
+    readonly columns = input.required<Column<T>[]>();
+    readonly columnBaseTranslationKey = input.required<string>();
+    readonly disabledIds = input<number[]>([]);
+
+    readonly onRowSelection = output<T>();
+
+    readonly isLoading = signal<boolean>(false);
+    private readonly searchResult = signal<SearchResult<T> | undefined>(undefined);
+    readonly resultsOnPage = computed(() => this.searchResult()?.resultsOnPage ?? []);
+
+    private readonly defaultSortingOrder = SortingOrder.ASCENDING;
+
+    readonly searchTerm = signal<string>('');
+    readonly sortingOrder = signal<SortingOrder>(this.defaultSortingOrder);
+    readonly sortedColumn = signal<string>('ID');
+    readonly page = signal<number>(1);
+    readonly pageSize = signal<number>(10).asReadonly();
+    readonly collectionSize = computed(() => {
+        const numberOfPages = this.searchResult()?.numberOfPages ?? 1;
+        return numberOfPages === 1 ? this.resultsOnPage().length : numberOfPages * this.pageSize();
+    });
```

Comment:
Could we get rid of the magic numbers? Especially as the component might be adapted and grow it helps to identify relations and makes the code easier to read 

```suggestion
    const PAGE_SIZE = 10;
    const MIN_NUMBER_OF_PAGES = 1;
    
    readonly pageSize = signal<number>(PAGE_SIZE).asReadonly();
    readonly collectionSize = computed(() => {
        const numberOfPages = this.searchResult()?.numberOfPages ?? MIN_NUMBER_OF_PAGES;
        return numberOfPages === MIN_NUMBER_OF_PAGES ? this.resultsOnPage().length : numberOfPages * this.pageSize();
    });
```

Reply:
For the page size I think it makes sense, but for the numberOfPages it wouldn't really add value I would say..

Reply:
From my personal experience, it helps if components grow, especially if many magic numbers are defined. It can be quite confusing which values are actually related to each other.

I do not insist on the variable for the min number of pages though